samtranslator/model/api/api_generator.py

```diff
@@ -184,8 +185,12 @@ def _construct_stage(self, deployment, swagger):
         # If StageName is some intrinsic function, then don't prefix the Stage's logical ID
         # This will NOT create duplicates because we allow only ONE stage per API resource
         stage_name_prefix = self.stage_name if isinstance(self.stage_name, string_types) else ""
-
-        stage = ApiGatewayStage(self.logical_id + stage_name_prefix + 'Stage',
+        if stage_name_prefix.isalnum():
+            stage_logical_id = self.logical_id + stage_name_prefix + 'Stage'
+        else:
+            generator = logical_id_generator.LogicalIdGenerator(self.logical_id + 'Stage', stage_name_prefix)
+            stage_logical_id = generator.gen()
```

Comment:
I think I like your approach more than something like `stage_logical_id = re.sub('[^A-Za-z0-9]+', '', stage_logical_id)` to remove all non-alphanumeric characters, as your approach differentiates between the stage names that you provided in your tests.